src/main/java/com/jozufozu/flywheel/compat/SodiumCompat.java

```diff
@@ -0,0 +1,14 @@
+package com.jozufozu.flywheel.compat;
+
+import com.jozufozu.flywheel.backend.instancing.InstancedRenderDispatcher;
+
+import net.caffeinemc.mods.sodium.api.blockentity.BlockEntityRenderHandler;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+
+public class SodiumCompat {
+	public static void forBlockEntityType(BlockEntityType<?> type) {
+		BlockEntityRenderHandler.instance().addRenderPredicate(type, (blockGetter, blockPos, blockEntity) ->
+				InstancedRenderDispatcher.tryAddBlockEntity(blockEntity)
+		);
+	}
```

Comment:
This method may be called multiple times for the same block entity type. I assume Sodium will keep adding new predicates. Confirm if this is the case, and if so, use a static set to keep track of which block entity types already have a registered predicate and don't add a new predicate if one has already been added previously. Also, move the last parenthesis to the line above it.

Reply:
I have confirmed that that is the case and that a predicate removal function exists. The best way to use the API is to modify the `setController` implementation in the `BlockEntityMixin`. If the current controller is null and the new controller is not null, add a predicate to Sodium's API and store the predicate in a field in the mixin. If the current controller is not null and the new controller is null, remove the predicate from Sodium's API and set the predicate field to null.